### PR TITLE
Make it possible to get more data, on more cells

### PR DESCRIPTION
**Pull request by jennybc**
This makes it possible to get full cell data, including formats, and to *not* drop empty cells (cells with a format, but no value).

### CODE REVIEW

**Comment by jennybc** [comment]
@nacnudus I'm going to merge this soon, but if you have any quick comments or reactions, feel free to share.

@Hadley I will merge this soon, but if you have a quick opinion re: argument names, please say so.

File: R/get_cells.R [hunk 1]
```diff
@@ -3,14 +3,18 @@ get_cells <- function(ss,
                       sheet = NULL,
                       range = NULL,
                       col_names_in_sheet = TRUE,
-                      skip = 0, n_max = Inf) {
+                      skip = 0, n_max = Inf,
+                      detail_level = c("default", "full"),
```
Comment (jennybc) [inline R/get_cells.R]: This is the user-facing argument that controls how much cell data to get:

  * `"default"` gets every field that googlesheets4 consults to guess or impose cell/column type
  * `"full"` gets all fields. It's meant for someone who wants to do artisanal [unpivotr-style](https://github.com/nacnudus/unpivotr) work that involves formats (in the sense of borders, backgrounds, etc.)

File: R/get_cells.R [hunk 2]
```diff
@@ -3,14 +3,18 @@ get_cells <- function(ss,
...
+                      discard_empty = TRUE) {
```
Comment (jennybc) [inline R/get_cells.R]: This controls whether we discard empty cells, i.e. cells that have no value. Usually will be `TRUE`, unless you really want to look at formats.

**Comment by nacnudus** [comment]
This is wonderful, thank you!

The example might be clearer if [deaths](https://docs.google.com/spreadsheets/d/1tuYKzSbLukDLe5ymf_ZKdQA8SfOyeMM7rmf6D6NJpxg/edit#gid=1210215306) had a more gaudilly formatted empty cell.

**Comment by jennybc** [comment]
OK the example is more exciting now (which also uncovered another bug) a6378922ed393ba0b1d2c42bfd1e049c67530e6c.

**Pull request merged**